.github/workflows/compilation_on_sgx.yml

```diff
@@ -290,6 +290,22 @@ jobs:
 
       - name: run spec tests
         run: |
+          set +e
           source /opt/intel/sgxsdk/environment
           ./test_wamr.sh ${{ matrix.test_option }} -t ${{ matrix.running_mode }}
+          exitcode="$?"
+
+          if [ $exitcode -eq 143 ]; then
+            echo "$exitcode is a known GitHub-hosted runner issue"
+            echo "::notice::Bypass an error with code 143 in SGX CI"
+            exit 0
```

Comment:
doesn't this risk us to merge broken changes?

Reply:
I believe it will not introduce any other unexpected breaking changes.

- Only bypass exit code 143. Other non-zero codes will still be captured.
- Spec_test scripts, all.py and runtest.py, will clamp return codes, and 143 is not one of them.
- 143 is not an errno in errno.h

Reply:
well, my concern is:

1. someone submits a PR with broken changes
2. the ci happened to pass because of this workaround
3. reviewers don't notice it and merge

Reply:
Based on my observations, if there is an error from the spec-test itself, 143 won't be generated. I believe the 143 code will not cover other errors. Those accidental 143 errors are quite annoying, so I want to relieve our daily work. If directly bypassing is too aggressive, how about automatically rerunning if it fails? 

- https://blog.logto.io/automatic-github-workflow-rerun
- https://github.com/actions/runner-images/issues/6680

Reply:
> Those accidental 143 errors are quite annoying, so I want to relieve our daily work.

i agree.

> If directly bypassing is too aggressive, how about automatically rerunning if it fails?

it seems nicer.